rust/private/rustfmt.bzl

```diff
@@ -0,0 +1,139 @@
+"""A module defining rustfmt rules"""
+
+load(":common.bzl", "rust_common")
+load(":rust.bzl", "rust_binary")
+load(":utils.bzl", "find_toolchain")
+
+def _rustfmt_aspect_impl(target, ctx):
+    if rust_common.crate_info not in target:
+        return []
+
+    crate_info = target[rust_common.crate_info]
+    manifest = ctx.actions.declare_file(ctx.label.name + ".rustfmt")
+    ctx.actions.write(
+        output = manifest,
+        content = "\n".join([
+            crate_info.edition,
+        ] + [src.path for src in crate_info.srcs.to_list() if not src.path.startswith("bazel-out")]),
+    )
+
+    return [
+        DefaultInfo(
+            files = depset([manifest]),
+        ),
+        OutputGroupInfo(
+            rustfmt = depset([manifest]),
+        ),
+    ]
+
+rustfmt_aspect = aspect(
+    doc = "",
+    implementation = _rustfmt_aspect_impl,
+)
+
+def _rustfmt_check_aspect_impl(target, ctx):
+    if rust_common.crate_info not in target:
+        return []
+
+    crate_info = target[rust_common.crate_info]
+
+    # Filter out any generated files
+    srcs = [src for src in crate_info.srcs.to_list() if not src.path.startswith("bazel-out")]
```

Comment:
https://docs.bazel.build/versions/master/skylark/lib/File.html#is_source should work there?

Reply:
Oh cool, thanks!